torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -349,17 +353,24 @@ struct GraphFuser {
     Node * in_graph = subgraph.createClone(n,[&](Value * k)-> Value* {
       return inputs_map[k];
     });
-    // if n is already an input to the fusion group,
-    // we need to remove it because n is now inside the fusion group
+    // if n's outputs are already inputs to the fusion group,
+    // we need to remove them because n is now inside the fusion group.
+    //
+    // i.e.,
+    // x = f(w); group(x, y, z) becomes group(w, y, z).
+    // x, y, z = f(w); group(x, y, z) becomes group(w).
+    //
     // remapping nodes that used the input to the newly-merged node
     // n is not an input when the fusion group is empty
     auto inputs = group->inputs();
-    auto it = std::find(inputs.begin(), inputs.end(), n->output());
-    if(it != inputs.end()) {
-      size_t p = it - inputs.begin();
-      group->removeInput(p);
-      subgraph.inputs()[p]->replaceAllUsesWith(in_graph->output());
-      subgraph.eraseInput(p);
+    for (size_t i = 0; i < n->outputs().size(); ++i) {
```

Comment:
why did this change? multiple outputs can happen due to chunk operators?

Reply:
Yes, this changed because chunk operators have multiple outputs